### PR TITLE
Make it possible to run E2E OS OAuth java selenium tests against Eclipse Che on QE OCP

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
Make it possible to run E2E OS OAuth java selenium tests against Eclipse Che on QE OCP.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17125

### CODE REVIEW

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[crw-ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[crw-ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

File: tests/legacy-e2e/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ocp/OpenShiftLoginPage.java [hunk 1]
```diff
@@ -66,6 +124,73 @@ public void waitOnOpen() {
...
+    } catch (TimeoutException e) {
+      return false;
+    }
+
+    return true;
+  }
+
+  public Boolean isOpenshiftLoginPageVisible() {
+    try {
+      seleniumWebDriverHelper.waitVisibility(By.id("kc-page-title"));
```
Comment (dmytro-ndp) [inline tests/legacy-e2e/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ocp/OpenShiftLoginPage.java]: It's worth `kc-page-title` to be put into **Locators** constants.

**Comment by dmytro-ndp** [comment]
@SkorikSergey: IMHO it would better to avoid changing all tests by putting 'openShiftLoginPage.login();' before `dashboard.open()` method.
Why don't move `openShiftLoginPage.login();` to [Dashboard#open() method](https://github.com/eclipse/che/blob/ef896437560b454218f8829f3bca4c5803f7b0dd/tests/legacy-e2e/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/Dashboard.java#L275)?

File: tests/.infra/centos-ci/nightly/ocp-oauth-test.sh [hunk 2]
```diff
@@ -34,14 +34,24 @@ installCheCtl
...
+
 bash tests/legacy-e2e/che-selenium-test/selenium-tests.sh \
-  --threads=1 \
+  --threads=3 \
   --host=${CHE_ROUTE} \
   --https \
   --port=443 \
   --multiuser \
   --fail-script-on-failed-tests \
-  --test=org.eclipse.che.selenium.site.ocpoauth.** \
```
Comment (dmytro-ndp) [inline tests/.infra/centos-ci/nightly/ocp-oauth-test.sh:44]: `org.eclipse.che.selenium.site.ocpoauth.**` tests are not being run by default. 

Did you ensure they were running after this change?

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Comment by SkorikSergey** [comment]
[ci-test]

**Review by dmytro-ndp — APPROVED** [review]
Looks good!
I would only ask to include all tests into oauth nightly tests execution https://github.com/eclipse/che/blob/master/tests/.infra/centos-ci/nightly/ocp-oauth-test.sh

**Comment by SkorikSergey** [comment]
[crw-ci-test]

**Comment by SkorikSergey** [comment]
[crw-ci-test]

**Comment by SkorikSergey** [comment]
[crw-ci-test]

**Pull request merged**